torch/_dynamo/variables/torch.py

```diff
@@ -1179,6 +1211,43 @@ def speculate_branch(branch):
             )
             r = body_r.as_proxy().node.meta["example_value"]
             example_value = r
+        elif self.value.__name__ in (
+            "trampoline_autograd_fwd",
+            "trampoline_autograd_bwd",
+            "trampoline_autograd_apply",
+        ):
+            always_restore = self.value.__name__ == "trampoline_autograd_bwd"
```

Comment:
Hmm, is the reason why we don't `always_restore=True` on trampoline_autograd_fwd is that we need to keep around the outputs so that we can speculate the backward?

Reply:
And because we want to install guards from it.